### PR TITLE
Widget Visibility: Disable in Block Editor

### PR DESCRIPTION
**Pull request by tyxla**
Currently, on `widget_admin_setup`, in the Widget Visibility module we'll run heavy operations to retrieve a bunch of entities, but we only really use those on the widgets page. Among those entities include all pages, categories, tags, custom taxonomy terms, all registered post types, users. For sites that don't leverage cache and have many pages with a lot of content, this can cause significant delays in TTFB and even end up in OOM errors.

The reason for this is that since Gutenberg 8.3 we call the `widget_admin_setup` hook on the block editor, too (see https://github.com/WordPress/gutenberg/pull/22807), but never actually use this data in the block editor. This means that we can disable the heavy logic on the block editor pages. This is what this PR does.

Note that this intentionally touches WP.com files. We'd like this improvement to be done there, too.

#### Changes proposed in this Pull Request:
* Widget Visibility: Disable in Block Editor

#### Jetpack product discussion
No related discussion. This is related to recent block editor perf improvements in p4TIVU-9uP-p2. See p7H4VZ-2KJ-p2 for more information.

#### Does this pull request change what data or activity we track or use?
Not at all.

#### Testing instructions:
* Go to the Widgets screen.
* Verify the Widget Visibility functionality still works well.

#### Proposed changelog entry for your changes:
* Disable Widget Visibility admin setup in Block Editor

### CODE REVIEW

**Comment by tyxla** [comment]
Not sure why the wpcom check reports tests as failed, but they actually passed.

**Review by jeherve — APPROVED** [review]
This should do the trick for now. 👍

**Comment by simison** [comment]
How does this behave with latest Gutenberg plugin and Block editor Widget screen enabled?

**Comment by tyxla** [comment]
> How does this behave with latest Gutenberg plugin and Block editor Widget screen enabled?

I don't think this screen is affected in any way. The new block editor widget screen is essentially a new page in the admin, and doesn't even call the `widget_admin_setup` hook or any other related hook that we run on the old-school widgets page.

**Comment by tyxla** [comment]
@jeherve thanks a bunch for the review 🙌  would be really appreciated if you could take a look at the corresponding diff, too. Thanks!

**Comment by simison** [comment]
FYI @adamziel in context of https://github.com/WordPress/gutenberg/issues/25836

File: modules/widget-visibility/widget-conditions.php
Comment (anomiex) [inline modules/widget-visibility/widget-conditions.php:26]: Can we just do a `function_exists()` or `is_callable()` check instead of having a todo comment? Or is someone committing to address this later?

Or is there concern that `wp_should_load_block_editor_scripts_and_styles()` isn't actually going to be added in WP 5.6 after all?

Comment (jeherve) [reply modules/widget-visibility/widget-conditions.php:26]: We usually keep those `TODO` items, alongside with a specific WordPress version when a replacement function gets added. This way, when Jetpack will start requiring WordPress 5.6+, we'll be able to go back and edit the codebase around all `TODO` comments that refer to that in the codebase.

You're right though, we could start using the function right now (it's been added already, see [related ticket](https://core.trac.wordpress.org/ticket/51330)), with a `function_exists()` check, this way this would make our job easier and faster whenever we have to come back when Jetpack will require 5.6+. :)

@tyxla What do you think?

Comment (tyxla) [reply modules/widget-visibility/widget-conditions.php:26]: Hey, folks! 

First of all, I did consider using the function directly, with a `function_exists` as you suggested. And if this was an instance where we don't have any alternative option, I likely would have used it that way.

However, the mechanism we're using here is totally safe and works in any modern WordPress. I'd say it's more stable if we keep it that way right now.

My rationale is that although the function is merged, it exists only in a development version of WordPress, and as such, the name of the function, or its functionality could change before the release. There is no guarantee this won't happen -  that's why we don't push features against non-stable code. 

So I'd very much prefer keeping this as-is, at least until the first RC (or the first beta at least). That will give us some time to react and update it.

How does that sound?

Comment (tyxla) [reply modules/widget-visibility/widget-conditions.php:26]: That being said, I'm not even sure if I still have merge permissions, so I'll leave this to you folks to decide 😉

**Review by jeherve — APPROVED** [review]

**Pull request merged**
